src/test/transaction_tests.cpp

```diff
@@ -228,16 +248,15 @@ BOOST_AUTO_TEST_CASE(tx_valid)
                 BOOST_ERROR("Bad test flags: " << strTest);
             }
 
-            if (!CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, ~verify_flags, txdata, strTest, /* expect_valid */ true)) {
-                BOOST_ERROR("Tx unexpectedly failed: " << strTest);
-            }
+            BOOST_CHECK_MESSAGE(CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, ~verify_flags, txdata, strTest, /* expect_valid */ true),
+                                "Tx unexpectedly failed: " << strTest);
 
             // Backwards compatibility of script verification flags: Removing any flag(s) should not invalidate a valid transaction
-            for (size_t i = 0; i < mapFlagNames.size(); ++i) {
+            for (const auto & [_, flag] : mapFlagNames) {
                 // Removing individual flags
-                unsigned int flags = TrimFlags(~(verify_flags | (1U << i)));
+                unsigned int flags = TrimFlags(~(verify_flags | flag));
                 if (!CheckTxScripts(tx, mapprevOutScriptPubKeys, mapprevOutValues, flags, txdata, strTest, /* expect_valid */ true)) {
-                    BOOST_ERROR("Tx unexpectedly failed with flag " << ToString(i) << " unset: " << strTest);
+                    BOOST_ERROR("Tx unexpectedly failed with flag " << ToString(flag) << " unset: " << strTest);
```

Comment:
This doesn't need `ToString()`:

```suggestion
                    BOOST_ERROR("Tx unexpectedly failed with flag " << flag << " unset: " << strTest);
```

or alternatively, print the name of the flag:

```suggestion
                    BOOST_ERROR("Tx unexpectedly failed with flag " << _ << " unset: " << strTest);
```

(but probably change `_` to `flag_name` or similar)

Reply:
printing name of flag! :D